From f389569ed5253690a2e3223b81d083566abe3e0e Mon Sep 17 00:00:00 2001
From: Guo Zihua <guozihua@huawei.com>
Date: Fri, 31 Dec 2021 15:59:53 +0800
Subject: [PATCH] KABI: reserve space for IMA IPE
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: f389569ed5253690a2e3223b81d083566abe3e0e


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4GK6B
CVE: NA
---------------------------

Reserving some fields for future IMA IPE development.

Signed-off-by: Guo Zihua <guozihua@huawei.com>
Reviewed-by: weiyang wang <wangweiyang2@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/blk_types.h | 2 ++
 include/linux/lsm_hooks.h | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/include/linux/blk_types.h b/include/linux/blk_types.h
index d9b69bbde5cc..f33725f03525 100644
--- a/include/linux/blk_types.h
+++ b/include/linux/blk_types.h
@@ -9,6 +9,7 @@
 #include <linux/types.h>
 #include <linux/bvec.h>
 #include <linux/ktime.h>
+#include <linux/kabi.h>
 
 struct bio_set;
 struct bio;
@@ -46,6 +47,7 @@ struct block_device {
 	int			bd_fsfreeze_count;
 	/* Mutex for freeze */
 	struct mutex		bd_fsfreeze_mutex;
+	KABI_RESERVE(1)
 } __randomize_layout;
 
 /*
diff --git a/include/linux/lsm_hooks.h b/include/linux/lsm_hooks.h
index a8531b37e6f5..4d49255a0921 100644
--- a/include/linux/lsm_hooks.h
+++ b/include/linux/lsm_hooks.h
@@ -28,6 +28,7 @@
 #include <linux/security.h>
 #include <linux/init.h>
 #include <linux/rculist.h>
+#include <linux/kabi.h>
 
 /**
  * union security_list_options - Linux Security Module hook function list
@@ -1566,6 +1567,7 @@ struct lsm_blob_sizes {
 	int	lbs_ipc;
 	int	lbs_msg_msg;
 	int	lbs_task;
+	KABI_RESERVE(1)
 };
 
 /*
-- 
2.26.2

